### DISCUSSION TITLE
().readline

### DISCUSSION THREAD

**Discussion by woojamon** [koka-lang/koka#447] [General]
Just curious, why does `"hello".println` compile but `().readline` does not? The error is "function is applied with too many arguments"

**Comment by TimWhiting** [comment]
Pretty simple:
`"hello".println` doesn't get translated to `println"hello"` but `println("hello")`.

So `().readline` gets translated to `readline(())` (1 argument of type unit) which isn't the same as `readline()` (0 arguments) which is what the function expects.

**Reply by woojamon** [reply]
Ah, I got bit by `unit`, `()`, and `(())` again, but that explanation makes sense. Thanks!
